niceml/data/datasets/dfdataset.py

```diff
@@ -27,51 +38,148 @@ class RegDataInfo(DataInfo):
     data: dict
 
     def get_info_dict(self) -> dict:
+        """
+        The get_info_dict function returns a dictionary containing the dataid
+        and all the data in self.data.
+
+        Returns:
+            A dictionary containing the dataid and all the other key-value pairs in self
+        """
         info_dict = dict(dataid=self.dataid)
         info_dict.update(self.data)
         return info_dict
 
     def get_identifier(self) -> str:
+        """
+        The get_identifier function returns the dataid of this object.
+
+        Returns:
+            The dataid
+        """
         return self.dataid
 
+    def __getattr__(self, item) -> Any:
+        """
+        The __getattr__ function is called when an attribute lookup has not found the attribute
+        in the usual places (i.e. it is not an instance attribute nor is it
+        found in the class tree for self).
+
+        Args:
+            item: Access the value of a key in the dictionary
+
+        Returns:
+            The value of the key in the data dictionary
+        """
+        return self.data[item]
+
 
 class DfDataset(Dataset, Sequence):  # pylint: disable=too-many-instance-attributes
     """Dataset for dataframes"""
 
-    def __init__(  # pylint: disable=too-many-arguments
+    def __init__(  # ruff: noqa: PLR0913
         self,
         id_key: str,
         batch_size: int,
         set_name: str,
-        df_loader: DfLoader,
-        df_path: str,
+        data_location: Union[dict, LocationConfig],
+        df_path: str = ExperimentFilenames.SUBSET_NAME,
         shuffle: bool = False,
+        data_shuffler: Optional[DataShuffler] = None,
+        dataframe_filters: Optional[List[DataframeFilter]] = None,
+        feature_combiners: Optional[List[FeatureCombiner]] = None,
+        extra_key_list: Optional[List[str]] = None,
     ):
+        """
+        The __init__ function is called when the class is instantiated.
+        It sets up the instance of the class, and defines all of its attributes.
+        The __init__ function takes in a number of arguments, which are used to set up
+        these attributes.
+
+        Args:
+            id_key: str: Specify the column name of the id column in your dataframe
+            batch_size: int: Define the size of each batch
+            set_name: str: Identify the dataset
+            data_location: Union[dict, LocationConfig]: Specify the location of the data
+            df_path: str: Specify the file name of the dataframe
```

Comment:
If 'df_path' is actually a file name and not a path, it should be renamed to 'data_filename' or something like that.

Reply:
Done